app/models/deleted_object.rb

```diff
@@ -23,4 +25,9 @@ class DeletedObject < ApplicationRecord
       & (created_at <= 1.week.ago)
     end
   end
+
+  def object_instance
+    obj_attrs = {id: object_id}.merge(metadata)
+    object_type.constantize.new(obj_attrs, without_protection: true)
```

Comment:
i would call `readonly!` on this instance, just to make it sure that it won't be saved..

Reply:
Right! Good one! 👍

Reply:
And even freeze it so internal variables won't be modified

Reply:
Good 👍 I will 👍